client/cpu_sched.cpp

```diff
@@ -1627,13 +1627,23 @@ ACTIVE_TASK* CLIENT_STATE::get_task(RESULT* rp) {
 void CLIENT_STATE::set_ncpus() {
     int ncpus_old = ncpus;
 
+    // config file can say to act like host has N CPUs
+    //
+    static bool first = true;
+    static int original_p_ncpus;
+    if (first) {
+        original_p_ncpus = host_info.p_ncpus;
+        first = false;
+    }
```

Comment:
I believe that these lines would be better implemented as simply:

`static int original_p_ncpus = host_info.p_ncpus;`

If I understand things correctly, a static function variable is allocated when the program starts but it is initialized the first time control passes through its initialization block.  See https://stackoverflow.com/questions/55510/when-do-function-level-static-variables-get-allocated-initialized for a discussion about this.